File: benchmarks/strings/string_interpolation.py
# String interpolation - Python
# Tests f-string performance

def generate_message(name, age, score):
    return f"Hello {name}, you are {age} years old and scored {score} points!"

# Generate many interpolated strings
total_length = 0

for i in range(50000):
    msg = generate_message("Alice", i, i * 2)
    total_length += len(msg)

print(total_length)

File: benchmarks/strings/string_interpolation.js
// String interpolation - JavaScript
// Tests template literal performance

function generateMessage(name, age, score) {
    return `Hello ${name}, you are ${age} years old and scored ${score} points!`;
}

// Generate many interpolated strings
let totalLength = 0;

for (let i = 0; i < 50000; i++) {
    const msg = generateMessage("Alice", i, i * 2);
    totalLength += msg.length;
}

console.log(totalLength);